django/db/models/expressions.py

```diff
@@ -536,6 +539,60 @@ def _prepare(self, output_field=None):
         return self
 
 
+class SlicableF(F):
+    """
+    An object that can add slicing notation to F expression.
+
+    Object resolves the column on which the slicing is applied, and then
+    applies the slicing if possible.
+    """
+    def __init__(self, f, k):
+        if not isinstance(k, (int, slice)):
+            raise TypeError
+        assert ((not isinstance(k, slice) and (k >= 0)) or
+                (isinstance(k, slice) and (k.start is None or k.start >= 0) and
+                 (k.stop is None or k.stop >= 0))), \
+            "Negative indexing is not supported."
+
+        assert ((not isinstance(k, slice)) or
+                (isinstance(k, slice) and k.step is None)), \
+            "Step argument is not supported."
+
+        self.expression = f
+        if isinstance(k, slice):
+            if k.start is not None:
+                self.low = int(k.start) + 1
+            else:
+                self.low = 1
+            if k.stop is not None:
+                self.length = int(k.stop) - self.low + 1
+            else:
+                self.length = None
+        else:
+            self.low = int(k)
+            self.length = 1
+
+    def __repr__(self):
+        return "{}({})".format(self.__class__.__name__, self.expression.name)
+
+    def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize=False, for_save=False):
+        text_fields = ['BinaryField', 'CharField', 'EmailField', 'FilePathField',
+                       'GenericIPAddressField', 'SlugField', 'TextField', 'URLField'
+                       ]
+        resolved = query.resolve_ref(self.expression.name, allow_joins, reuse, summarize)
+        if resolved.target.__class__.__name__ in text_fields:
```

Comment:
I think `get_internal_type` is better to use.

Reply:
Agreed. Didn't think of it. Fixed that.